modules/gateway/nodes.go

```diff
@@ -53,14 +54,9 @@ func (g *Gateway) managedAddUntrustedNode(addr modules.NetAddress) error {
 	//
 	// NOTE: this is a somewhat clunky way of specifying that you didn't
 	// actually want a connection.
-	encoding.WriteObject(conn, "0.0.0")
-	var reject string
-	err = encoding.ReadObject(conn, &reject, build.MaxEncodedVersionLength)
-	if err != nil {
-		g.log.Debugln("ERROR: version handshake ping terminated unexpectedly:", err)
-	}
-	if reject != "reject" {
-		g.log.Debugln("WARN: peer does not seem to have correctly rejected our ping:", reject)
+	_, err = connectVersionHandshake(conn, "0.0.0")
+	if err != errPeerRejectedConn {
```

Comment:
do we know that the rejection is because of the version? If so, the variable name should probably be made more specific.

I guess that if they are returning a Sia error string, they are probably an active Sia node, so this should be good enough.